docs/contributing/versioning.md

```diff
@@ -0,0 +1,102 @@
+# Versioning
+
+With our user interface (UI) library, users [can consume our code in different ways](#public-api).
+
+We follow [Semantic Versioning](https://semver.org/) but a UI library often has subjective changes such as visual spacing changes.
+
+This document aims to outline the processes we follow when we version our releases.
+
+## Stability
+
+A stable library prioritises the users and ecosystem that it supports.
+
+We often release new components and features which will encourage people to update.
+
+But we should make sure that we only make breaking changes when we have a good reason, and have decided that it is absolutely necessary.
+
+Good examples of such situations would be:
+
+- issues that are barriers for end-users (users of services) based on evidence (for example user research)
+- issues that are barriers for users (users of GOV.UK Frontend) based on evidence
+- accessibility issues
+- security issues
+- performance issues
+
+We would not make breaking changes for:
+
+- you think you can sneak a change in with other breaking changes
```

Comment:
perhaps replace "you think you can" with "thinking you can" for sake of consistent use of gerunds in this list